.travis.yml

```diff
@@ -25,15 +27,25 @@ cache:
     - $VERILATOR_CACHE
 
 before_install:
-# Perl modules needed for testing
-# Not listing Bit::Vector as slow to install, and only skips one test
-  - touch temp.cpp ; g++ -E -dM -c temp.cpp | sort ; rm -rf temp.cpp
-  - yes yes | sudo cpan -fi Unix::Processors Parallel::Forker
-  - sudo apt-get install gdb gtkwave lcov
-  - sudo apt-get install libfl-dev || true
-  - sudo apt-get install libgoogle-perftools-dev
-# Only works on 20.04
-  - sudo apt-get install libsystemc-dev || true
+  - |
+    if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+      touch temp.cpp ; g++ -E -dM -c temp.cpp | sort ; rm -rf temp.cpp;
+      sudo apt-get install gdb gtkwave lcov;
+      sudo apt-get install libfl-dev || true;
+      sudo apt-get install libgoogle-perftools-dev;
+      # Only works on 20.04
+      sudo apt-get install libsystemc-dev || true;
+    fi
+  - |
+    if [ "$TRAVIS_OS_NAME" = "freebsd" ]; then
+      sudo pkg install -y autoconf bison ccache gmake perl5 python3;
```

Comment:
I think installing ccache here should be enough to run the tests fast enough (the build certainly seems to use ccache). I am under the imporession that we don't actually persis the ccache directory, but it speeds up tests as each test tends to rebuild some common files (i.e.: ccache reuse is only within a single test run)

Reply:
That is, even with the travis cacher issue.

Reply:
Note Travis does persist the ccache (when you use their ccache, not install it yourself).

Reply:
I might be wrong here but looking at the cache -s at the bottom of tests you seem to get pretty much the same stats with a 30 something % hit rate for all runs of the same job. If the cache was persistent I would expect patches like this to get 100% as the compiler is not actually changed. Also at the top when it sets up the cache it only adds verilator_cache but not .ccache. See also at the bottom it reporting not saving cache due to no changes. Maybe specifying the cache directories section (to include verilator_cache) overrides the default of adding the .ccache?

Reply:
https://travis-ci.com/github/gezalore/verilator/jobs/342381593#L198